client/rest-high-level/src/main/java/org/elasticsearch/client/tasks/CancelTasksResponse.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.client.tasks;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
+import org.elasticsearch.common.xcontent.XContentParser;
+
+import static java.util.stream.Collectors.groupingBy;
+import static java.util.stream.Collectors.toList;
+import static org.elasticsearch.common.xcontent.ConstructingObjectParser.optionalConstructorArg;
+
+public class CancelTasksResponse {
+
+    private final NodesInfoData nodesInfoData;
+    private final List<TaskOperationFailure> taskFailures = new ArrayList<>();
+    private final List<ElasticsearchException> nodeFailures = new ArrayList<>();
+    private final List<TaskInfo> tasks = new ArrayList<>();
+    private final List<TaskGroup> taskGroups = new ArrayList<>();
+
+    public CancelTasksResponse(NodesInfoData nodesInfoData,
+                               List<TaskOperationFailure> taskFailures,
+                               List<ElasticsearchException> nodeFailures) {
+        this.nodesInfoData = nodesInfoData;
+        if (taskFailures!= null){
+            this.taskFailures.addAll(taskFailures);
+        }
+        if (nodeFailures!=null) {
+            this.nodeFailures.addAll(nodeFailures);
+        }
+        this.tasks.addAll(nodesInfoData
+            .getNodesInfoData()
+            .stream()
+            .flatMap(nodeData -> nodeData.getTasks().stream())
+            .collect(toList())
+        );
+
+        this.taskGroups.addAll(buildTaskGroups());
+    }
+
+    private List<TaskGroup> buildTaskGroups() {
```

Comment:
I think all of this, including the classes you have created for this can be removed. This was only used in the CAT api on the server side, so we can just return the same thing the transport response returns.

Reply:
so shall I remove the TaskGroup from this API?

Reply:
https://www.elastic.co/guide/en/elasticsearch/client/java-rest/current/java-rest-high-cluster-cancel-tasks.html

they are present in the API, do they need to be removed?